### PR TITLE
doesn't store reference in memory if local

### PR DESCRIPTION
**Pull request by erictu**
Also fixes bugs in isLocal

### CODE REVIEW

**Comment by erictu** [comment]
Jenkins, test this please.

**Comment by erictu** [comment]
Jenkins, test this please

**Comment by erictu** [comment]
Jenkins, test the please.

**Comment by erictu** [comment]
Jenkins, test this please.

File: mango-core/src/main/scala/org/bdgenomics/mango/util/ResourceUtils.scala [hunk 1]
```diff
@@ -53,12 +53,12 @@ object ResourceUtils {
...
   def isLocal(filePath: String, sc: SparkContext): Boolean = {
-    val localFile: File = new File(filePath)
     val path: Path = new Path(filePath)
     val fs: FileSystem = path.getFileSystem(sc.hadoopConfiguration)
-    if (localFile.exists) {
+    val homedir = fs.getHomeDirectory.toString
+    if (homedir.startsWith("file:") && fs.exists(path)) {
       true
-    } else if (fs.exists(path)) {
+    } else if (homedir.startsWith("hdfs:/") && fs.exists(path)) {
```
Comment (akmorrow13) [inline mango-core/src/main/scala/org/bdgenomics/mango/util/ResourceUtils.scala]: What about support for amazon s3?

**Comment by erictu** [comment]
Jenkins, test this please.

**Comment by erictu** [comment]
Jenkins, test this please.

**Comment by erictu** [comment]
Jenkins, retest this please.

**Comment by erictu** [comment]
Jenkins, retest this please

**Pull request merged**